homeassistant/helpers/entity_component.py

```diff
@@ -295,9 +293,40 @@ def __init__(self, component, platform, scan_interval, entity_namespace):
         self.scan_interval = scan_interval
         self.entity_namespace = entity_namespace
         self.platform_entities = []
+        self._tasks = []
         self._async_unsub_polling = None
         self._process_updates = asyncio.Lock(loop=component.hass.loop)
 
+    @asyncio.coroutine
+    def async_block_entities_done(self):
+        """Wait until all entities add to hass."""
+        if self._tasks:
+            pending = [task for task in self._tasks if not task.done()]
+            self._tasks.clear()
+
+            if pending:
+                yield from asyncio.wait(pending, loop=self.component.hass.loop)
+
+    def add_platform_entities(self, new_entities, update_before_add=False):
+        """Add entities for a single platform."""
+        if update_before_add:
+            for entity in new_entities:
+                entity.update()
+
+        run_callback_threadsafe(
+            self.component.hass.loop,
+            self.async_add_platform_entities, list(new_entities), False
+        ).result()
+
+    @callback
+    def async_add_platform_entities(self, new_entities,
```

Comment:
We should name this better. Just adding platform is confusing because it is on a platform. 

I think that we should call this async_add_entities and rename the other one to _async_add_entities. 

What shall we do with all the task objects? Maybe it should not append a task object to the list if it is running?